common/util/FileUtil.cpp

```diff
@@ -130,6 +127,17 @@ std::string get_current_executable_path() {
 #endif
 }
 
+std::string get_mod_name() {
+  //Returns the name of the folder that gk is in, useful for detecting name of mod for unique saves and other implmentations.
```

Comment:
in development though the binary will be like `out/build/Release/bin/gk.exe`, so won't you end up with `bin` as the mod name? not the end of the world but maybe a bit confusing... I'm not sure what's the best way to avoid having some sort of MOD_NAME_PLEASE_CHANGE_THIS somewhere...

Reply:
Yeah, in development source everything would still share saves like it currently does in the `bin` folder.

Reply:
> I'm not sure what's the best way to avoid having some sort of MOD_NAME_PLEASE_CHANGE_THIS somewhere...

imo, this isn't a bad thing -- but you obviously want it somewhere sensible.  It's far better for a mod to provide a name than for the tooling to just try to assume or figure it out.

The most obvious reasons that comes to mind are imagine if two mods have the same name / if the folder structures of the mods change / etc.

- In the short-term, this could just be a CLI arg that the launcher(s) provide.
- In the long-term, mods would have their own project file that could set metadata like this.

Reply:
I kinda liked it just being dependent on the name of the folder myself, but yeah we could add some metadata somewhere I guess. I mostly just named this function badly. Its more get_folder_name() and not `mod_name` for certain things and having one source of truth yeah falling back to a json/metadata in ONE spot is good.